src/levanter/models/flash_attention.py

```diff
@@ -307,6 +307,11 @@ def do_inner_block(state):
             dV_j = dV_j + hax.dot(QPos.name, p_ij, dO_i).astype(dV_j.dtype)
             dK_j = dK_j + hax.dot(QPos.name, dAttn_ij, q_i).astype(dK_j.dtype)
 
+            # GQA-specific: eliminate unnecessary axes (e.g. 'q_heads_per_group')
+            unnecessary_axes = hax.eliminate_axes(dV_j.axes, v.axes)
+            dV_j = hax.sum(dV_j, unnecessary_axes)
+            dK_j = hax.sum(dK_j, unnecessary_axes)
```

Comment:
I'm not sure if this is a good place to do the summation. If there's a better place please let me know. I guess the placement of this logic would affect performance of FA. @dlwh

Reply:
wait, is this the right semantics? I think, if nothing else, you need to do this reduction before you add it to the prior dV_j/d_K_j

Reply:
You're right. My newest revision should work. I did just randomly pick the variable names for the intermediate dot product result. let me know if you want some other meaningful variable names for them.